src/librustc_metadata/decoder.rs

```diff
@@ -398,7 +399,7 @@ pub fn get_adt_def<'tcx>(intr: &IdentInterner,
                 did: did,
                 name: item_name(intr, item),
                 fields: get_variant_fields(intr, cdata, item, tcx),
-                disr_val: disr,
+                disr_val: ConstInt::Infer(disr),
```

Comment:
not sure how to get the value of the correct type, I could serialize it along with the value, but it must be encoded somewhere in the enum, too, right?